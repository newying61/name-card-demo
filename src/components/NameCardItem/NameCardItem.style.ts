import styled from 'styled-components';

export const StyledDd = styled.dd`
  word-break: break-all;
`;