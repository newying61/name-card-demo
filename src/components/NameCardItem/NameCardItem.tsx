import React from 'react';
import ListGroup from 'react-bootstrap/ListGroup';
import { StyledLabel, StyledValueSpan } from './NameCardItem.style';

interface NameCardItemProps {
  fieldName: string;
  fieldValue: string;
}

const NameCardItem: React.FC<NameCardItemProps> = ({ fieldName, fieldValue }) => {
  return (
    <ListGroup.Item as="li">
      <StyledLabel>{fieldName}</StyledLabel>
      <StyledValueSpan>{fieldValue}</StyledValueSpan>
    </ListGroup.Item>
  );
}

export default NameCardItem;