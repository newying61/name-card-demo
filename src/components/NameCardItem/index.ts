import NameCardItem from './NameCardItem';

export default NameCardItem;
