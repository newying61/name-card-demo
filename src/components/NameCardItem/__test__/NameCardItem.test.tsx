import React from 'react';
import { render, cleanup } from '@testing-library/react';
import NameCardItem from '../';

afterEach(cleanup);

describe('NameCardItem', () => {
    it('should render without crash', () => {
      const { container } = render(
        <NameCardItem fieldName="test" fieldValue="test" />
      );

      expect(container.firstChild).toMatchSnapshot();
  });
});