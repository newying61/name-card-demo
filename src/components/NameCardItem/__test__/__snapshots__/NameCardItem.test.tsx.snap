// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameCardItem should render without crash 1`] = `
<li
  class="list-group-item"
>
  <dt>
    test
  </dt>
  <dd
    class="sc-bdVaJa bNqPsR"
  >
    test
  </dd>
</li>
`;
